pkg/reconcile/pipeline/handler/collect/impl_test.go

```diff
@@ -1030,6 +1030,71 @@ var _ = Describe("Integration Collect definitions + items", func() {
 
 })
 
+var _ = Describe("Handle invalid sliceOfMap annotations", func() {
+	var (
+		mockCtrl *gomock.Controller
+	)
+
+	BeforeEach(func() {
+		mockCtrl = gomock.NewController(GinkgoT())
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+	})
+
+	type testCase struct {
+		serviceContent map[string]interface{}
+	}
+
+	It("should set an error condition",
+		func() {
+			tc := testCase{
+				serviceContent: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"annotations": map[string]interface{}{
+							"service.binding/java-maven_port": "path={.status.ports},elementType=sliceOfMaps,sourceKey=name,sourceValue=asdf",
+						},
+					},
+					"status": map[string]interface{}{
+						"ports": []interface{}{
+							map[string]interface{}{"name": "foo", "value": "bar"},
+						},
+					},
+				},
+			}
+
+			ctx := mocks.NewMockContext(mockCtrl)
+			service := mocks.NewMockService(mockCtrl)
+			serviceResource := &unstructured.Unstructured{Object: tc.serviceContent}
+
+			ctx.EXPECT().Services().Return([]pipeline.Service{service}, nil).MinTimes(1)
+
+			var bindingDefs []binding.Definition
+			service.EXPECT().AddBindingDef(gomock.Any()).DoAndReturn(func(bd binding.Definition) { bindingDefs = append(bindingDefs, bd) }).Times(len(serviceResource.GetAnnotations()))
+			service.EXPECT().BindingDefs().DoAndReturn(func() []binding.Definition { return bindingDefs })
+			service.EXPECT().Resource().Return(serviceResource).Times(2)
+
+			crd := mocks.NewMockCRD(mockCtrl)
+			crd.EXPECT().Descriptor().Return(nil, nil)
+			crd.EXPECT().Resource().Return(&unstructured.Unstructured{})
+			service.EXPECT().CustomResourceDefinition().Return(crd, nil)
+
+			ctx.EXPECT().SetCondition(
+				apis.Conditions().NotCollectionReady().
+					Reason(collect.ValueNotFound).
+					Msg("Value for key java-maven_port_foo not found").
+					Build())
+			ctx.EXPECT().Error(collect.ErrorValueNotFound)
+			ctx.EXPECT().StopProcessing()
+
+			collect.BindingDefinitions(ctx)
```

Comment:
this test should check the behavior of `BindingItems` stage only, we do not need perform an integration test here. Let's reduce the test to unit test by dropping this line and modifying the setup accordingly.